### PR TITLE
Parameters support

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

File: examples/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
 		"url": "git://github.com/redislabs/redisgraph.js.git"
 	},
 	"dependencies": {
-		"redisgraph.js": "^1.1.3"
+		"redisgraph.js": "git://github.com/RedisGraph/redisgraph.js.git"
```
Comment (gkorland) [inline examples/package.json]: should we set it to ^1.2.0

File: src/graph.js [hunk 2]
```diff
@@ -44,15 +44,52 @@ class Graph {
...
+            let strValue = "";
+            if (paramValue[0] != "\"")
+                strValue += "\"";
+            strValue += paramValue;
+            if(paramValue[paramValue.length-1] != "\"")
+                strValue += "\"";
+            return strValue;
+        }
+        if(Array.isArray(paramValue))
+            return "[" + paramValue.toString() + "]";
```
Comment (gkorland) [inline src/graph.js]: should every paramValue be checked as String and escaped?

File: src/graph.js [hunk 3]
```diff
@@ -44,15 +44,52 @@ class Graph {
...
+     * Extracts parameters from dictionary into cypher parameters string.
+     * 
+     * @param params parameters dictionary.
+     * @return a cypher parameters string.
+     */
+    buildParamsHeader(params) {
+        let paramsHeader = "CYPHER ";
+        for (var key in params) {
+            let value = this.paramToString(params[key]);
+            paramsHeader += `${key}=${value} `;
```
Comment (gkorland) [inline src/graph.js]: Perhaps it's more efficient to use array.push() and then join() like StringBuilder in Java

File: src/graph.js [hunk 4]
```diff
@@ -44,15 +44,52 @@ class Graph {
...
 	/**
 	 * Execute a Cypher query (async)
 	 *
 	 * @param query Cypher query
+     * @param params Parameters map
 	 * @return a promise contains a result set
 	 */
-	async query(query) {
+	async query(query, params) {
+        if(params){
```
Comment (gkorland) [inline src/graph.js]: check for arguments.len()>1 
```
if(params)
```
returns false also in case params==null

**Comment by gkorland** [comment]
@DvirDukhan please add negative tests too

File: src/graph.js (hunk 3 above)
Comment (DvirDukhan) [reply src/graph.js]: done

File: src/graph.js (hunk 2 above)
Comment (DvirDukhan) [reply src/graph.js]: added test

File: examples/package.json (hunk 1 above)
Comment (DvirDukhan) [reply examples/package.json]: done

File: src/graph.js [hunk 5]
```diff
@@ -59,8 +59,13 @@ class Graph {
...
             return strValue;
         }
-        if(Array.isArray(paramValue))
-            return "[" + paramValue.toString() + "]";
+        if(Array.isArray(paramValue)) {
+            let stringsArr = new Array(paramValue.length);
+            for(var i = 0; i < paramValue.length; i++) {
+                stringsArr[i] = this.paramToString(paramValue[i]);
+            }
+            return "[" + stringsArr.join(", ") + "]";
```
Comment (gkorland) [inline src/graph.js]: Why not avoid this + also

Comment (DvirDukhan) [reply src/graph.js]: fixed

Comment (gkorland) [reply src/graph.js]: @DvirDukhan sorry my mistake I didn't notice the "," in the join I think your original code is better.

**Comment by bionicles** [comment]
Would it be possible to document this in the README?

**Pull request merged**